src/combinatorics/simple_graph/strongly_regular.lean

```diff
@@ -41,27 +40,133 @@ A graph is strongly regular with parameters `n k l m` if
+  (⊥ : simple_graph V).is_SRG_with (fintype.card V) 0 l 0 :=
+{ card := rfl,
+  regular := bot_degree,
+  adj_common := λ v w h, h.elim,
+  nadj_common := λ v w h, begin
+    simp only [finset.card_eq_zero, finset.filter_congr_decidable, fintype.card_of_finset],
+    ext,
+    simp [mem_common_neighbors],
+  end }
 
-/-- Complete graphs are strongly regular. Note that the parameter `m` can take any value
-  for complete graphs, since there are no distinct pairs of nonadjacent vertices. -/
-lemma complete_strongly_regular (m : ℕ) :
-  (⊤ : simple_graph V).is_SRG_of (fintype.card V) (fintype.card V - 1) (fintype.card V - 2) m :=
+/-- Complete graphs are strongly regular. Note that `m` can take any value
+  for complete graphs, since there are no distinct pairs of non-adjacent vertices. -/
+lemma top_strongly_regular :
+  (⊤ : simple_graph V).is_SRG_with (fintype.card V) (fintype.card V - 1) (fintype.card V - 2) m :=
 { card := rfl,
-  regular := complete_graph_degree,
-  adj_common := λ v w (h : v ≠ w),
-    begin
-      simp only [fintype.card_of_finset, mem_common_neighbors, filter_not, ←not_or_distrib,
-                 filter_eq, filter_or, card_univ_diff, mem_univ, if_pos, ←insert_eq, top_adj],
-      rw [card_insert_of_not_mem, card_singleton],
-      simp [h]
-    end,
-  nadj_common := λ v w (h : ¬(v ≠ w) ∧ _), (h.1 h.2).elim }
+  regular := complete_graph_is_regular,
+  adj_common := λ v w h, begin
+    rw card_common_neighbors_top,
+    exact h,
+  end,
+  nadj_common := λ v w h, false.elim $ by simpa using h }
+
+lemma card_neighbor_finset_union_eq {v w : V} (h : G.is_SRG_with n k l m) :
+  (G.neighbor_finset v ∪ G.neighbor_finset w).card =
+    2 * k - fintype.card (G.common_neighbors v w) :=
+begin
+  apply @nat.add_right_cancel _ (fintype.card (G.common_neighbors v w)),
+  rw [nat.sub_add_cancel, ← set.to_finset_card],
+  { simp [neighbor_finset, common_neighbors, set.to_finset_inter, finset.card_union_add_card_inter,
+      G.is_regular_of_degree_eq h.regular, two_mul], },
+  { apply le_trans (card_common_neighbors_le_degree_left _ _ _),
+    simp [G.is_regular_of_degree_eq h.regular, two_mul], },
+end
+
+/-- Assuming `G` is strongly regular, `2*(k + 1) - m` in `G` is the number of vertices that are
+  adjacent to either `v` or `w` when `¬G.adj v w`. So it's the cardinality of
+  `G.neighbor_set v ∪ G.neighbor_set w`. -/
+lemma card_neighbor_finset_union_nadj {v w : V} (h : G.is_SRG_with n k l m)
+  (hne : v ≠ w) (ha : ¬G.adj v w) :
+  (G.neighbor_finset v ∪ G.neighbor_finset w).card = 2 * k - m :=
+begin
+  rw ← h.nadj_common v w ⟨hne, ha⟩,
+  apply G.card_neighbor_finset_union_eq h,
+end
+
+lemma card_neighbor_finset_union_adj {v w : V} (h : G.is_SRG_with n k l m)
+  (ha : G.adj v w) :
+  (G.neighbor_finset v ∪ G.neighbor_finset w).card = 2 * k - l :=
+begin
+  rw ← h.adj_common v w ha,
+  apply G.card_neighbor_finset_union_eq h,
+end
+
+@[simp] theorem finset_compl_union (s t : finset V) : (s ∪ t)ᶜ = sᶜ ∩ tᶜ := compl_sup
```

Comment:
Please move this to `data/fintype/basic` as `finset.compl_union`. May as well add `finset.compl_inter` while you're at it.